tests/model/liveselection.js

```diff
@@ -872,4 +872,37 @@ describe( 'LiveSelection', () => {
 			expect( values ).to.deep.equal( [] );
 		} );
 	} );
+
+	describe( '_updateAttributes()', () => {
```

Comment:
Besides, there are many tests missing. E.g. `<p>x[<$text bold=true>bar</$text><img></img>foo]</p>` and similar where object is inside the selection (or in a place reachable by that algorithm) but should not affect the method's behaviour.